source/plugin/Assets/GoogleMobileAds/Platforms/Unity/RewardedAdClient.cs

```diff
@@ -0,0 +1,205 @@
+
+using System.Reflection;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using GoogleMobileAds;
+using GoogleMobileAds.Api;
+using GoogleMobileAds.Common;
+
+namespace GoogleMobileAds.Unity
+{
+    public class RewardedAdClient : BaseAdDummyClient, IRewardedAdClient
+    {
+        // Ad event fired when the rewarded ad has been received.
+        public event EventHandler<EventArgs> OnAdLoaded;
+        // Ad event fired when the rewarded ad has failed to load.
+        public event EventHandler<AdErrorEventArgs> OnAdFailedToLoad;
+        // Ad event fired when the rewarded ad has failed to show.
+        public event EventHandler<AdErrorEventArgs> OnAdFailedToShow;
+        // Ad event fired when the rewarded ad is opened.
+        public event EventHandler<EventArgs> OnAdOpening;
+        // Ad event fired when the rewarded ad has rewarded the user.
+        public event EventHandler<Reward> OnUserEarnedReward;
+        // Ad event fired when the rewarded ad is closed.
+        public event EventHandler<EventArgs> OnAdClosed;
+        // Ad event fired when the rewarded ad is estimated to have earned money.
+        public event EventHandler<AdValueEventArgs> OnPaidEvent;
+
+        private static readonly Dictionary<AdSize, string> prefabAds = new Dictionary<AdSize, string>()
+        {
+            {new AdSize (320,480), "DummyAds/Rewarded/320x480" },
+            {new AdSize (480,320), "DummyAds/Rewarded/480x320"},
+            {new AdSize (768,1024), "DummyAds/Rewarded/768x1024" },
+            {new AdSize (1024,768), "DummyAds/Rewarded/1024x768"}
+        };
+
+        private ButtonBehaviour buttonBehaviour;
+
+        private void AddClickBehavior(GameObject dummyAd)
+        {
+            Image myImage = dummyAd.GetComponentInChildren<Image>();
+            Button button = myImage.GetComponentInChildren<Button>();
+            button.onClick.AddListener(() =>
+            {
+                buttonBehaviour.OpenURL();
+            });
+            Button[] innerButtons = dummyAd.GetComponentsInChildren<Button>();
+
+            innerButtons[1].onClick.AddListener(() =>
+            {
+                AdBehaviour.DestroyAd(dummyAd);
+                prefabAd = null;
+                if (OnAdClosed != null)
+                {
+                    OnAdClosed.Invoke(this, EventArgs.Empty);
+                }
+                AdBehaviour.ResumeGame();
+                if (OnUserEarnedReward != null)
+                {
+                    OnUserEarnedReward.Invoke(this, GetRewardItem());
+                }
+            });
+        }
+
+        private void CreateButtonBehavior()
+        {
+            buttonBehaviour = new ButtonBehaviour();
+            buttonBehaviour.OnAdOpening += OnAdOpening;
+        }
+
+        // Creates a rewarded ad.
+        public void CreateRewardedAd(string adUnitId)
+        {
+            Debug.Log("Dummy " + MethodBase.GetCurrentMethod().Name);
```

Comment:
We will want to not have these comments since this isn't the "dummy" client anymore. Please remove.